### PR TITLE
O3-2862 - Queue Module - validate that queue entry falls within visit…

### PR DESCRIPTION
**Pull request by mseaton**
… when saving visit

### CODE REVIEW

**Review by ibacher — APPROVED** [review]

**Review by ibacher — APPROVED** [review]

File: api/src/main/java/org/openmrs/module/queue/validators/VisitWithQueueEntriesValidator.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		}
+		Visit visit = (Visit) target;
+		
+		// This implementation is copied to match the core validator approach to nested encounters
+		if (visit.getId() != null) {
+			Date startDateTime = visit.getStartDatetime();
+			Date stopDateTime = visit.getStopDatetime();
+			
+			QueueEntrySearchCriteria criteria = new QueueEntrySearchCriteria();
+			criteria.setIsEnded(null);
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/queue/validators/VisitWithQueueEntriesValidator.java:53]: I assume setting this to null means returns all queues, both those that are ended and those that aren't ended.

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @mseaton ... nice that you can add in validators for existing objects this way.

File: api/src/main/java/org/openmrs/module/queue/validators/VisitWithQueueEntriesValidator.java (hunk 1 above)
Comment (mseaton) [reply api/src/main/java/org/openmrs/module/queue/validators/VisitWithQueueEntriesValidator.java:53]: Correct, the default (which I don't love, but the way it is implemented) is to only return active entries by default.  So to get all entries, you need to explicitly null out this property :/

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @mseaton

**Pull request merged**